File: Server/src/main/java/core/game/content/ame/events/genie/GenieNPC.java
package core.game.content.ame.events.genie;

import core.game.content.ame.AntiMacroNPC;
import core.game.node.entity.player.Player;
import core.game.world.map.Location;
import core.game.world.update.flag.context.Graphics;

/**
 * Handles the random event genie npc.
 * @author Vexia
 */
public final class GenieNPC extends AntiMacroNPC {

	/**
	 * The force chats to pronounce.
	 */
	private static final String[] CHATS = new String[] { "Greetings, @gL @name!", "Ehem... @gender @name?", "Are you there, @gL @name?", "No one ignores me!" };

	/**
	 * Constructs a new {@code GenieNPC} {@code Object}.
	 * @param id the id.
	 * @param location the location.
	 * @param event the event.
	 * @param player the player.
	 */
	public GenieNPC(int id, Location location, GenieRandomEvent event, Player player) {
		super(id, location, event, player, CHATS);
	}

	/**
	 * Constructs a new {@code GenieNPC} {@code Object}.
	 */
	public GenieNPC() {
		super(0, null, null, null);
	}

	@Override
	public void init() {
		super.init();
		graphics(new Graphics(86));
	}

	@Override
	public int[] getIds() {
		return new int[] { 409 };
	}

}

File: Server/src/main/kotlin/rs09/game/content/ame/events/genie/GenieNPC.kt
package rs09.game.content.ame.events.genie

import core.game.node.entity.npc.NPC
import core.tools.RandomFunction
import org.rs09.consts.Items
import org.rs09.consts.NPCs
import rs09.game.content.ame.RandomEventNPC
import rs09.game.content.global.WeightBasedTable

class GenieNPC(override var loot: WeightBasedTable? = null) : RandomEventNPC(NPCs.GENIE_409) {
    val phrases = arrayOf("Greetings, @name!","Ehem... Master @name?","Are you there, Master @name?","No one ignores me!")
    var assigned_item = 0
    val items = arrayOf(Items.LAMP_2528)

    override fun tick() {
        if(RandomFunction.random(1,15) == 5){
            sendChat(phrases.random().replace("@name",player.name.capitalize()))
        }
        super.tick()
    }

    override fun init() {
        super.init()
        assignItem()
        sendChat(phrases.random().replace("@name",player.name.capitalize()))
    }

    fun assignItem(){
        assigned_item = items.random()
        player.setAttribute("genie:item",assigned_item)
    }

    override fun talkTo(npc: NPC) {
        player.dialogueInterpreter.open(GenieDialogue(),npc)
    }
}